File: documentapi/src/vespa/documentapi/messagebus/messages/emptybucketsmessage.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "emptybucketsmessage.h"

namespace documentapi {

EmptyBucketsMessage::EmptyBucketsMessage() :
    _bucketIds()
{
}

EmptyBucketsMessage::EmptyBucketsMessage(const std::vector<document::BucketId> &bucketIds) :
    _bucketIds(bucketIds)
{
}

EmptyBucketsMessage::~EmptyBucketsMessage() = default;

void
EmptyBucketsMessage::setBucketIds(std::vector<document::BucketId> bucketIds)
{
    _bucketIds = std::move(bucketIds);
}

void
EmptyBucketsMessage::resize(uint32_t size)
{
    _bucketIds.resize(size);
}

DocumentReply::UP
EmptyBucketsMessage::doCreateReply() const
{
    return DocumentReply::UP(new VisitorReply(DocumentProtocol::REPLY_EMPTYBUCKETS));
}

uint32_t
EmptyBucketsMessage::getType() const
{
    return DocumentProtocol::MESSAGE_EMPTYBUCKETS;
}

}

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/EmptyBucketsMessage.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.protocol;

import com.yahoo.document.BucketId;

import java.util.ArrayList;
import java.util.List;

/**
 * @author banino
 */
public class EmptyBucketsMessage extends VisitorMessage {

    private final List<BucketId> bids = new ArrayList<BucketId>();

    EmptyBucketsMessage() {
        // must be deserialized into
    }

    public EmptyBucketsMessage(List<BucketId> bids) {
        this.bids.addAll(bids);
    }

    public List<BucketId> getBucketIds() {
        return bids;
    }

    public void setBucketIds(List<BucketId> bids) {
        this.bids.clear();
        this.bids.addAll(bids);
    }

    @Override
    public DocumentReply createReply() {
        return new VisitorReply(DocumentProtocol.REPLY_EMPTYBUCKETS);
    }

    @Override
    public int getType() {
        return DocumentProtocol.MESSAGE_EMPTYBUCKETS;
    }
}
